### PR TITLE
[Bug] zos_operator throws TypeError when wait_time_s is specified 

### LINKED ISSUE #377: [Enhancement] Validate timeout function delivered by ZOAU
At the time of development this `timout` parameter was not implemented although conveyed so it must be tested after the latest zoau ASM updates. 

See file [zos_operator.py](plugins/modules/zos_operator.py)
See lines [298](https://github.com/ansible-collections/ibm_zos_core/pull/348/files#diff-28649d089de0e43c15567d3b7f14bb305d97dbce20f276e9676af687c398dee6R298) 
```
    if params.get("wait"):
        wait = params.get("wait_time_s")
        if wait:
            kwargs.update({"timeout": "{0}".format(wait)})
            kwargs.update({"parameters": "ISFDELAY={0}".format(wait)})
            # it *appears* IFSdelay is passing through correctly... did 1x-4x tests 0 to 20 seconds
```

### LINKED ISSUE #389: [Bug] zos_operator throws TypeError when wait_time_s is specified using ZOAU 1.2.1h
##### SUMMARY
When specifying wait_time_s in zos_operator using ZOAU 1.2.1h early version it throws TypeError.

##### COMPONENT NAME

zos_operator

##### ANSIBLE VERSION

```
Ansible version output:
ansible [core 2.11.12] 
  config file = None
  configured module search path = ['/Users/fernandoflores/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/fernandoflores/Library/Python/3.8/lib/python/site-packages/ansible
  ansible collection location = /Users/fernandoflores/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/fernandoflores/Library/Python/3.8/bin/ansible
  python version = 3.8.9 (default, Apr 13 2022, 08:48:07) [Clang 13.1.6 (clang-1316.0.21.2.5)]
  jinja version = 3.1.2
  libyaml = True

```

##### SPECIFY ANSIBLE COLLECTION VERSION

1.4.0-beta.1

##### SPECIFY THE Z OPEN AUTOMATION UTILITIES VERSION

2022/08/25 21:44:21 CUT V1.2.1 31163ab 1856

##### STEPS TO REPRODUCE

```yaml
Reproduction playbook/yaml
- hosts: zvm
  collections:
    - ibm.ibm_zos_core
  gather_facts: no
  vars:
    ZOAU: "/zoau/v1.2.1h"
    PYZ: "/python2/usr/lpp/IBM/cyp/v3r9/pyz"
  tasks:
    - name: Call opercmd with 5 seconds delay
      zos_operator:
        cmd: "d u,all"
        verbose: True
        wait_time_s: 5
        wait: True
      register: result

    - name: show  response
      debug: 
        msg: "{{ result }}"
```

##### EXPECTED RESULTS

Executes and returns the result with a wait of maximum 5 seconds for the job.

##### ACTUAL RESULTS

```
Playbook output:

fatal: [zvm]: FAILED! => {"changed": false, "msg": "An unexpected error occurred: ZOAUException(\"'timeout' must be an integer.\")"}
```

#### Ansible.cfg

```
'ansible-config dump --only-changed' output:
```

#### Inventory

```
Inventory content:
```

#### Vars

```
'group_vars' or 'host_vars' content:
```

### PR DESCRIPTION
**Pull request by fernandofloresg**
##### SUMMARY

After ZOAU implemented timeout option in opercmd support was missing from zos_operator. Changed the opercmd.execute_command call to use the latest interface and explicitly provide the timeout value, had to be done this way since zoau python utils does a validation in this value over kwargs. 
Added the case when wait_time_s is 0 to default to 1 and be true to the docs "When set to 0, the system default is used".

Also changed the test to now validate for timeouts and enabled the quick delay test.

**Update, in addition to the original issue these updates are have included in this PR:**
- bugfixes:
  - ibm_zos_operator - fixed such that specifying wait_time_s would throw an error (original issue in this PR)
  - ibm_zos_operator - fixed case sensitive error checks, invalid, error & unidentifiable 
  - ibm_zos_operator - fixed the wait_time_s to default to 1 second 
- minor_changes:
  - ibm_zos_operator - deprecated the wait option, not needed with wait_time_s minor_changes
  - ibm_zos_operator - added in the response the cmd result 
  - ibm_zos_operator - added in the response the wait_time_s set 
  - ibm_zos_operator - added in the response the elapsed time 
  

fixes #389 
fixes #377 
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

zos_operator

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by fernandofloresg** [comment]
Screenshot 2022-08-30 at 15 51 55
Screenshot 2022-08-30 at 15 52 43

File: plugins/modules/zos_operator.py [hunk 1]
```diff
@@ -294,16 +295,20 @@ def run_operator_command(params):
     if params.get("debug"):
```
[the remark this answers is not in the document]
Comment (ddimatos) [reply plugins/modules/zos_operator.py]: I have resolved this in the last commit

File: plugins/modules/zos_operator.py [hunk 2]
```diff
@@ -294,16 +295,20 @@ def run_operator_command(params):
...
-        wait = params.get("wait_time_s")
-        if wait:
-            kwargs.update({"timeout": "{0}".format(wait)})
-            kwargs.update({"parameters": "ISFDELAY={0}".format(wait)})
+        wait_s = params.get("wait_time_s")
+        # To comply with docs "When set to 0, the system default is used."
+        if wait_s == 0:
+            wait_s = 1
+        if wait_s:
+            # kwargs.update({"timeout": "{0}".format(wait_s)})
```
[the remark this answers is not in the document]
Comment (ddimatos) [reply plugins/modules/zos_operator.py]: This is cleaned up and resolved.

File: plugins/modules/zos_operator.py [hunk 3]
```diff
@@ -294,16 +295,20 @@ def run_operator_command(params):
...
+            kwargs.update({"parameters": "ISFDELAY={0}".format(wait_s)})
             # it *appears* IFSdelay is passing through correctly... did 1x-4x tests 0 to 20 seconds
     cmdtxt = params.get("cmd")
 
     args = []
-    rc, stdout, stderr = execute_command(cmdtxt, *args, **kwargs)
+    rc, stdout, stderr = execute_command(cmdtxt, wait_s, *args, **kwargs)
 
     extrastdout = ""
     if params.get("verbose"):
```
[the remark this answers is not in the document]
Comment (ddimatos) [reply plugins/modules/zos_operator.py]: This is done

**Comment by ddimatos** [comment]
Updated changes return a result (below), what is new in the response is the `wait_time_s`, `cmd` and `elapsed`.

```
ok: [zvm] => {
    "result": {
        "changed": true,
        "cmd": "d u,all",
        "content": [
            "EC33017A   2022244  22:14:53.00             ISF031I CONSOLE OMVS0000 ACTIVATED",
            "EC33017A   2022244  22:14:53.00            -D U,ALL ",
            "EC33017A   2022244  22:14:53.00             IEE457I 22.14.53 UNIT STATUS 067",
            "                                           UNIT TYPE STATUS        VOLSER     VOLSTATE      SS",
            "                                           0000 3390 F-NRD                        /RSDNT     0",
            "                                           0001 3211 OFFLINE                                 0",
            "                                           0002 3211 OFFLINE                                 0",
            "                                           0003 3211 OFFLINE                                 0",
            "                                           0004 3211 OFFLINE                                 0",
            "                                           0005 3211 OFFLINE                                 0",
            "                                           0006 3211 OFFLINE                                 0",
            "                                           0007 3211 OFFLINE                                 0",
            "                                           0008 3211 OFFLINE                                 0",
            "                                           0009 3277 OFFLINE                                 0",
            "                                           000C 2540 A                                       0",
            "                                           000D 2540 A                                       0",
            "                                           000E 1403 A                                       0",
            "                                           000F 1403 A                                       0",
            "                                           0010 3211 A                                       0",
            "                                           0011 3211 A                                       0",
            "",
            ""
        ],
        "deprecations": [
            {
                "collection_name": "ibm.ibm_zos_core",
                "date": "2022-09-01",
                "msg": "Support for configuring wait has been deprecated.Configuring wait is now managed by setting 'wait_time_s'"
            }
        ],
        "elapsed": 0.52,
        "failed": false,
        "rc": 0,
        "wait_time_s": 1
    }
}
```

**Comment by ddimatos** [comment]
Regression with updated responses:

image

100% regression success, [internal link/log](http://sprinkles1.fyre.ibm.com:8080/job/zosAnsible/1134/consoleText)

For the long running command which is commented out, we need a MVS command that can take a while to run, they all average 1/2 - 1 second , not long enough to evaluate. I do have a long running command I used that I not to be publicly shared that I tested with and it all ran fine. I will look for a valid command we can share by discussing it with some of the system admins.

**Comment by ddimatos** [comment]
Also addresses #377

**Review by fernandofloresg — COMMENTED** [review]
Looks good to me, this refactor removes unnecessary steps and simplifies the verbose and wait logic.

**Review by ddimatos — APPROVED** [review]

**Pull request merged**